src/commons/asBaseComponent.tsx

```diff
@@ -76,6 +76,10 @@ function asBaseComponent<PROPS, STATICS = {}>(WrappedComponent: React.ComponentT
   BaseComponent.displayName = WrappedComponent.displayName;
   BaseComponent.propTypes = WrappedComponent.propTypes;
   BaseComponent.defaultProps = WrappedComponent.defaultProps;
+  const ThemeContext = ThemeManager.getThemeContext();
+  if (ThemeContext) {
```

Comment:
1. Do we need the `if`?
2. Do we want to account for `WrappedComponent.contextType`?

Reply:
It felt safer with if, also typescript didn't like when I just assigned it 

What do you mean be account for? You mean as a breaking change? 
I believe we are the only one using the contextType

Reply:
I meant that maybe add something like:
```
else if (WrappedComponent.contextType) {
  BaseComponent.contextType = WrappedComponent.contextType;
```

Approving anyway